bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/internal/DPIZoomChangeRegistry.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Yatta Solutions and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Yatta Solutions - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.internal;
+
+import java.util.*;
+import java.util.Map.*;
+
+import org.eclipse.swt.widgets.*;
+
+public class DPIZoomChangeRegistry {
+
+	private static Map<Class<? extends Widget>, DPIZoomChangeHandler> dpiZoomChangeHandlers =  new TreeMap<>(
+			(o1, o2) -> {
+	            if(o1.isAssignableFrom(o2)) {
+	            	return -1;
+	            }
+	            if(o2.isAssignableFrom(o1)) {
+	            	return 1;
+	            }
+	            return o1.getName().compareTo(o2.getName());
+	        });
+
+	public static <E> void applyChange(Widget widget, int newZoom, float scalingFactor) {
```

Comment:
Add the JavaDoc here and mention explicitly that this method may end up invoking several `DPIZoomChangeHandler`s (mention the order in which they will be invoked too). This is an important detail.